src/model/range.js

```diff
@@ -462,6 +462,27 @@ export default class Range {
 				// <p>foo</p><p>[]bar{}</p> -> <p>foo[]bar{}</p>
 				ranges[ 0 ] = new Range( targetPosition.getShiftedBy( this.start.offset ) );
 			}
+			// Don't ask. Just debug.
+			// Like this: https://github.com/ckeditor/ckeditor5-engine/issues/841#issuecomment-282706488.
+			else if ( type == 'move' && deltaType == 'wrap' ) {
+				// <p>a[b</p><w></w><p>c]d</p> -> <w><p>a[b</p></w><p>c]d</p>
```

Comment:
It would be nice to expand this example a little bit more. State before wrapping, which element is wrapped, what conversion without the fix would look like, etc. (why we are taking start/end of exactly those ranges).